Pass teleport_version explicitly to AppUpdater

avatus opened the pull request:
Because the process.env was not mocked in the tests, these tests would fail locally if you had TELEPORT_TOOLS_VERSION=off in your env vars locally (CI doesnt have this so they dont fail, this is just for local). This PR will remove that env var from a mock env var so they pass regardless of computer running them

ravicious commented on web/packages/teleterm/src/services/appUpdater/appUpdater.test.ts:
```diff
@@ -38,7 +43,10 @@ jest.mock('electron', () => ({
+const originalEnv = process.env;
 beforeAll(() => {
+  process.env = { ...originalEnv };
+  process.env[TELEPORT_TOOLS_VERSION_ENV_VAR] = undefined;
```
We had this problem with `agentDownloader` before. https://github.com/gravitational/teleport/pull/28698#discussion_r1258292222 [`c3d8c0947`](https://github.com/gravitational/teleport/pull/28698/commits/c3d8c0947664a13d56723139439ddce368dc931f)

Instead of modifying the global value of `process.env`, it's better to make `AppUpdater` accept `env` in the constructor so that in tests we have total control over how the env var looks like. It's also more future proof in case other env vars are added later.

---

Regarding my comment in that PR:

> As far as I know, this is not safe because what if another test also sets `process.env` at the same time? Instead of reading `process.env` in `downloadAgent` directly, you can pass it as an argument so that you don't have to interact with `process.env` in tests.

This would only be a problem when another test _within the same file_ modifies `process.env`, as AFAIK Jest runs each file in a separate process environment (I think?).

This should be true because at the moment we do stuff like this and nothing appears to be blowing up ;f

https://github.com/gravitational/teleport/blob/70241501220c688091ffab5c4091e8ff3818b908/web/packages/teleterm/src/mainProcess/agentDownloader/agentDownloader.test.ts#L74-L77

avatus committed:
Pass teleport_version explicitly to AppUpdater

This passes the env var as a value rather than the entire process.env to
the AppUpdater constructor. This allows us to control the tests better
rather than relying on local environments

ravicious commented on web/packages/teleterm/src/services/appUpdater/appUpdater.ts:
```diff
@@ -100,6 +100,7 @@ export class AppUpdater {
     this.nativeUpdater.logger = this.logger;
     this.nativeUpdater.allowDowngrade = true;
     this.nativeUpdater.autoDownload = false;
+    this.versionEnvVar = versionEnvVar;
```
This one isn't actually needed, notice how we don't have an equivalent lines for `storage` or `getClusterVersions`. Grzegorz taught me this a long time ago.

https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd

avatus replied:
oh wow, interesting. i thought even what i had was cool defining in the constructor cuz i didnt even know that. i was doing the oldschool way of defining the type and setting in the constructor. double TIL

avatus commented:
@gzdunek how far to backport this? app updater is only v18 yes?

gzdunek reviewed (approved):
We also have the updater on v17.

Thanks for the fix!
